### PR TITLE
[INLONG-9867][Manager] Unified configuration process for standalone and sortflink

### LINKED ISSUE #9867: [Improve][Manager] Unified configuration process for standalone and sortflink
### Description

Unified configuration process for standalone and sortflink.

### InLong Component

InLong Manager

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by fuweng11**
### Prepare a Pull Request

- Fixes #9867 9867

### Motivation

Unified configuration process for standalone and sortflink.
The current configuration process of standalone has the following issues:
1. Configuration changes will be updated in real-time to the manager's cache. Once a configuration change occurs, sortstandalone will sense it and write data based on the new configuration.
Once the configuration is modified incorrectly, it can lead to issues with the entire write.

2. The configuration in the manager cache is to fully pull the content from the database and filter it. Each refresh requires a full database query and repeated filtering operations at the service layer.

3. There is no management function for whether each configuration is functioning properly. Once the configuration is registered on the manager side, it defaults to the correct configuration.
Failure to verify and issue each configuration results in unreliable configurations.

### Modifications
The flowchart of standalone pull configuration is as follows:
image

The getClusterConfigV2 interface has been set up, where the data comes from the sort_config table, which can only be obtained through workflow configuration operations. The interface protocol is consistent with the getClusterConfig interface.

1. Incorporate the configuration process of sort standalone into the manager's configuration management system and add a configuration information table. The JSON format configuration information that sort standalone needs to pull will be saved in a field of this table. Only those that are successfully configured will be recorded in this table.

2. Modify the configuration timer to be pulled to the cache and change it to actively trigger the configuration update cache. After successful configuration, the configuration information will be registered in this configuration information table, loaded into the cache, and the md5 value will be modified.

### CODE REVIEW

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/sort/SortStandAloneConfigOperator.java [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+                sortConfigEntity.setSourceParams(objectMapper.writeValueAsString(topic));
+                sortConfigEntity.setClusterParams(objectMapper.writeValueAsString(params));
+                sortConfigEntity.setInlongClusterTag(groupInfo.getInlongClusterTag());
+                sortConfigEntityMapper.updateByIdSelective(sortConfigEntity);
+            }
+        } catch (Exception e) {
+            LOGGER.error("fail to parse id params of groupId={}, streamId={} name={}, type={}",
+                    sinkEntity.getInlongGroupId(), sinkEntity.getInlongStreamId(),
+                    sinkEntity.getSinkName(), sinkEntity.getSinkType(), e);
+            throw new BusinessException("faild to save sort config for sinkId" + sinkEntity.getId());
```
Comment (aloyszhang) [inline inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/sort/SortStandAloneConfigOperator.java:175]:
```suggestion
            throw new BusinessException("failed to save sort config for sinkId" + sinkEntity.getId());
```

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/sort/SortStandAloneConfigOperator.java [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+            String namespace = sortGroupInfo.getMqResource();
+            String topic = streamEntity.getMqResource();
+            String fullTopic = tenant + InlongConstants.SLASH + namespace + InlongConstants.SLASH + topic;
+            Map<String, String> groupExt = sortGroupInfo.getExtParamsMap();
+            String groupTenant = Optional
+                    .ofNullable(groupExt.get(KEY_NEW_TENANT))
+                    .orElse(groupExt.get(KEY_OLD_TENANT));
+            if (StringUtils.isNotBlank(groupTenant)) {
+                fullTopic = groupTenant + InlongConstants.SLASH + namespace + InlongConstants.SLASH + topic;
+            }
```
Comment (aloyszhang) [inline inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/sort/SortStandAloneConfigOperator.java:198]:
```suggestion
           String namespace = sortGroupInfo.getMqResource();
            String topic = streamEntity.getMqResource();
            Map<String, String> groupExt = sortGroupInfo.getExtParamsMap();
            String groupTenant = Optional.ofNullable(groupExt.get(KEY_NEW_TENANT)).orElse(groupExt.get(KEY_OLD_TENANT));
            String tenant = StringUtils.isNotBlank(groupTenant)
                    ? groupTenant : Optional.ofNullable(param.get(KEY_NEW_TENANT)).orElse(param.get(KEY_OLD_TENANT));
            String fullTopic = tenant + InlongConstants.SLASH + namespace + InlongConstants.SLASH + topic;
```

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/core/SortClusterService.java [hunk 3]
```diff
@@ -32,4 +32,6 @@ public interface SortClusterService {
      * @return Corresponding response.
      */
     SortClusterResponse getClusterConfig(String clusterName, String md5);
+
```
Comment (aloyszhang) [inline inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/core/SortClusterService.java:35]: It's better to add docs for new added interface.

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/core/impl/SortClusterServiceImpl.java [hunk 4]
```diff
@@ -83,10 +87,15 @@ public class SortClusterServiceImpl implements SortClusterService {
...
     private Map<String, String> sortClusterMd5Map = new ConcurrentHashMap<>();
+    private Map<String, String> sortClusterMd5MapV2 = new ConcurrentHashMap<>();
+
     // key : sort cluster name, value : cluster config
     private Map<String, SortClusterConfig> sortClusterConfigMap = new ConcurrentHashMap<>();
+    private Map<String, SortClusterConfig> sortClusterConfigMapV2 = new ConcurrentHashMap<>();
+
     // key : sort cluster name, value : error log
     private Map<String, String> sortClusterErrorLogMap = new ConcurrentHashMap<>();
+    private Map<String, String> sortClusterErrorLogMapV2 = new ConcurrentHashMap<>();
```
Comment (aloyszhang) [inline inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/core/impl/SortClusterServiceImpl.java:98]: This map is not used anywhere.

Comment (fuweng11) [reply inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/core/impl/SortClusterServiceImpl.java:98]: Done.

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/core/SortClusterService.java (hunk 3 above)
Comment (fuweng11) [reply inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/core/SortClusterService.java:35]: Done.

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/sort/SortStandAloneConfigOperator.java (hunk 2 above)
Comment (fuweng11) [reply inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/sort/SortStandAloneConfigOperator.java:201]: Done.

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/sort/SortStandAloneConfigOperator.java (hunk 1 above)
Comment (fuweng11) [reply inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/sort/SortStandAloneConfigOperator.java:178]: Fixed.

**Review by aloyszhang — DISMISSED** [review]

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/core/SortService.java [hunk 5]
```diff
@@ -45,6 +45,22 @@ public interface SortService {
      */
     SortClusterResponse getClusterConfig(String clusterName, String md5);
 
+    /**
+     * Get sort cluster config V@.
```
[the remark this answers is not in the document]
Comment (fuweng11) [reply inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/core/SortService.java:49]: Done.

**Review by aloyszhang — DISMISSED** [review]

File: inlong-manager/manager-common/src/main/java/org/apache/inlong/manager/common/enums/ProcessName.java [hunk 6]
```diff
@@ -70,7 +70,9 @@ public enum ProcessName {
     /**
      * Delete inlong stream process
      */
-    DELETE_STREAM_RESOURCE("Delete Stream");
+    DELETE_STREAM_RESOURCE("Delete Stream"),
+
+    CREATE_CLUSTER_RESOURCE("Create Cluster");
```
Comment (aloyszhang) [inline inlong-manager/manager-common/src/main/java/org/apache/inlong/manager/common/enums/ProcessName.java:75]: Plz add dos

File: inlong-common/src/main/java/org/apache/inlong/common/pojo/sortstandalone/SortConfigResponse.java [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+@Data
+@Builder
+@NoArgsConstructor
+@AllArgsConstructor
+public class SortConfigResponse {
+
+    public static final int SUCC = 0;
+    public static final int NOUPDATE = 1;
+    public static final int FAIL = -1;
+    public static final int REQ_PARAMS_ERROR = -101;
```
Comment (aloyszhang) [inline inlong-common/src/main/java/org/apache/inlong/common/pojo/sortstandalone/SortConfigResponse.java:34]:
```suggestion
    public static final int NO_UPDATE = 1;
    public static final int FAIL = -1;
    public static final int REQUEST_PARAMS_ERROR = -101;
```

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/core/impl/SortServiceImpl.java [hunk 8]
```diff
@@ -53,6 +81,10 @@
 @Service
 public class SortServiceImpl implements SortService, PluginBinder {
 
+    private static final int RESPONSE_CODE_SUCCESS = 0;
+    private static final int RESPONSE_CODE_NO_UPDATE = 1;
+    private static final int RESPONSE_CODE_FAIL = -1;
+    private static final int RESPONSE_CODE_REQ_PARAMS_ERROR = -101;
```
Comment (aloyszhang) [inline inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/core/impl/SortServiceImpl.java:87]:
```suggestion
    private static final int RESPONSE_CODE_REQUEST_PARAMS_ERROR = -101;
```

File: inlong-common/src/main/java/org/apache/inlong/common/pojo/sortstandalone/SortConfigResponse.java (hunk 7 above)
Comment (fuweng11) [reply inlong-common/src/main/java/org/apache/inlong/common/pojo/sortstandalone/SortConfigResponse.java]: Done.

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/core/impl/SortServiceImpl.java [hunk 9]
```diff
@@ -117,4 +189,126 @@ public void acceptPlugin(Plugin plugin) {
             sortPoller = pollerPlugin.getSortPoller();
         }
     }
+
+    @Override
+    public SortConfigResponse getSortConfig(String clusterName, String md5) {
+        if (StringUtils.isBlank(clusterName)) {
+            String errMsg = "blank cluster name or task name, return nothing";
```
[the remark this answers is not in the document]
Comment (fuweng11) [reply inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/core/impl/SortServiceImpl.java]: Done.

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/core/impl/SortServiceImpl.java (hunk 8 above)
Comment (fuweng11) [reply inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/core/impl/SortServiceImpl.java]: Done.

File: inlong-manager/manager-common/src/main/java/org/apache/inlong/manager/common/enums/ProcessName.java (hunk 6 above)
Comment (fuweng11) [reply inlong-manager/manager-common/src/main/java/org/apache/inlong/manager/common/enums/ProcessName.java:78]: Done.

**Review by aloyszhang — APPROVED** [review]

**Review by aloyszhang — APPROVED** [review]
LGTM

**Review by dockerzhang — APPROVED** [review]

**Pull request merged**